### DISCUSSION TITLE
Help transitioning existing actions to block based checkout experience

### DISCUSSION THREAD

**Discussion by northernbeacheswebsites** [woocommerce/woocommerce#43678] [Checkout Flow]
Hi there, 

I am a developer of a plugin which uses the action: woocommerce_after_order_notes to add new fields to the checkout using: woocommerce_form_field. How do I get this working with the block based checkout experience as easily as possible?

I ideally just want to find the new hook where I can insert all my existing fields but I don't know if that is possible anymore. Some general guidance would be much appreciated.

**Comment by senadir** [comment]
Keep an eye on this discussion as we're soon to land methods to add new fields

https://github.com/woocommerce/woocommerce-blocks/discussions/11173

For now, you can [follow this tutorial](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md), but the above solution would make it much easier.

**Comment by northernbeacheswebsites** [comment]
Ok great, thank you :)
